packages/apidom-parser-adapter-yaml-1-2/src/syntactic-analysis/indirect/visitors/YamlAstVisitor.ts

```diff
@@ -39,190 +39,193 @@ export const keyMap = {
-      // @ts-ignore
-      sourceMap.position = node.position;
-      // @ts-ignore
-      sourceMap.astNode = node;
-      element.meta.set('sourceMap', sourceMap);
-    };
+  public processedDocumentCount: number;
 
-    /**
-     * Public API.
-     */
+  public annotations: AnnotationElement[];
 
-    this.namespace = createNamespace();
-    this.annotations = [];
+  public namespace: Namespace;
 
-    this.stream = {
-      leave(node: YamlStream) {
-        const element = new ParseResultElement();
-        // @ts-ignore
-        element._content = node.children.flat(1);
-
-        // mark first-non Annotation element as result
-        // @ts-ignore
-        const elements = element.findElements(isPrimitiveElement);
-        if (elements.length > 0) {
-          const resultElement = elements[0];
-          resultElement.classes.push('result');
-        }
-
-        // provide annotations
-        this.annotations.forEach((annotationElement: AnnotationElement) => {
-          element.push(annotationElement);
-        });
-        this.annotations = [];
-
-        return element;
-      },
-    };
-
-    this.comment = function comment(node: YamlComment) {
-      const isStreamComment = this.processedDocumentCount === 0;
-
-      // we're only interested of stream comments before the first document
-      if (isStreamComment) {
-        // @ts-ignore
-        const element = new CommentElement(node.content);
-        maybeAddSourceMap(node, element);
-        return element;
-      }
+  public stream = {
+    leave: (node: YamlStream): ParseResultElement => {
+      const element = new ParseResultElement();
+      // @ts-ignore
+      element._content = node.children.flat(1);
 
-      return null;
-    };
-
-    this.document = function document(node: YamlDocument) {
-      const shouldWarnAboutMoreDocuments = this.processedDocumentCount === 1;
-      const shouldSkipVisitingMoreDocuments = this.processedDocumentCount >= 1;
-
-      if (shouldWarnAboutMoreDocuments) {
-        const message =
-          'Only first document within YAML stream will be used. Rest will be discarded.';
-        const element = new AnnotationElement(message);
-        element.classes.push('warning');
-        maybeAddSourceMap(node, element);
-        this.annotations.push(element);
+      // mark first-non Annotation element as result
+      // @ts-ignore
+      const elements = element.findElements(isPrimitiveElement);
+      if (elements.length > 0) {
+        const resultElement = elements[0];
+        resultElement.classes.push('result');
       }
 
-      if (shouldSkipVisitingMoreDocuments) {
-        return null;
-      }
+      // provide annotations
+      this.annotations.forEach((annotationElement: AnnotationElement) => {
+        element.push(annotationElement);
+      });
+      this.annotations = [];
 
-      this.processedDocumentCount += 1;
+      return element;
+    },
+  };
 
-      return node.children;
-    };
+  constructor() {
```

Comment:
You can drop the sourceMap and processedDocumentCount initialization as they are initialized directly by property initializer.
```suggestion
  constructor() {
    this.annotations = [];
    this.namespace = createNamespace();
  }
```